models/restaurants.go

```diff
@@ -75,3 +76,30 @@ func OneRestaurant(id string) (Restaurant, error) {
 
 	return rt, nil
 }
+
+// UpdateOneRestaurant will access database to update the Restaurant by id
+func UpdateOneRestaurant(id string, body map[string]string) error {
+	var args []interface{}
+	sqlString := "UPDATE restaurants SET"
+	for k, v := range body {
+		sqlString += " " + k + "=?,"
+		args = append(args, v)
+	}
+	sqlString = strings.TrimSuffix(sqlString, ",") + " WHERE id=?"
```

Comment:
Why not just use placeholders?
```suggestion
	stmtIns, err = db.Prepare(`UPDATE restaurants 
	SET name=?, address=?, number=?
	WHERE id=?`) 
	defer stmtIns.Close()
	
	_, err = stmtIns.Exec(body["name"], body["address"], body["number"])
```

I think we should to use a restaurant object rather than a map here but maybe another time.

Reply:
The reason why I used map because I want to update the columns want to be changed.

Reply:
We should be able to use `COALESCE` 
```SQL
UPDATE restaurants
SET name=COALESCE(?, name)
address = COALESCE(?, address)
number = COALESCE(?, number)
WHERE id=?
```
https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html

Reply:
@kroyagis  I tried this, and because the struct has been set the default values, the columns will be reset

Reply:
the type just had to be changed the value type to pointer `*string` so that it can hold `nil`